internal/backends/mcu/pico_st7789.rs

```diff
@@ -47,11 +50,43 @@ type IrqPin = gpio::Pin<gpio::bank0::Gpio17, gpio::PullUpInput>;
 static IRQ_PIN: Mutex<RefCell<Option<IrqPin>>> = Mutex::new(RefCell::new(None));
 
 static ALARM0: Mutex<RefCell<Option<Alarm0>>> = Mutex::new(RefCell::new(None));
+static TIMER: Mutex<RefCell<Option<Timer>>> = Mutex::new(RefCell::new(None));
 
 // 16ns for serial clock cycle (write), page 43 of https://www.waveshare.com/w/upload/a/ae/ST7789_Datasheet.pdf
 const SPI_ST7789VW_MAX_FREQ: embedded_time::rate::Hertz = embedded_time::rate::Hertz(62_500_000u32);
 
+const DISPLAY_SIZE: PhysicalSize = PhysicalSize::new(320, 240);
+
 pub fn init() {
+    unsafe { ALLOCATOR.init(&mut HEAP as *const u8 as usize, core::mem::size_of_val(&HEAP)) }
+    i_slint_core::backend::instance_or_init(|| alloc::boxed::Box::new(PicoBackend));
+}
+
+thread_local! { static WINDOW: RefCell<Option<Rc<PicoWindow>>> = RefCell::new(None) }
+static NEEDS_REDRAW: AtomicBool = AtomicBool::new(false);
+
+struct PicoBackend;
+impl i_slint_core::backend::Backend for PicoBackend {
+    fn create_window(&self) -> i_slint_core::api::Window {
+        i_slint_core::window::WindowInner::new(|window| {
+            Rc::new(PicoWindow { self_weak: window.clone(), renderer: Default::default() })
```

Comment:
Perhaps if the `PicoWindow` was also created inside `run_event_loop`, the `NEEDS_REDRAW` and `WINDOW` global would not be needed. Just a thought, not needed for this change obviously.

Reply:
Yeah, but the window needs to be created before the component, which is created before we run the event loop.

Reply:
Yes, the `i_slint_core::api::Window` cannot be created before that, but the `Rc<PicoWindow>` could?

I see that it as two fields, the renderer and the self_weak, and I think both are okay to create earlier?